### ISSUE TITLE
Workstation 16 support

### ISSUE THREAD

**Issue by VladislavAntonyuk** [paolo-projects/auto-unlocker#26]

**Comment by julianxhokaxhiu** [comment]
There's no need. V1.1.1 works just fine and you can use macOS up to v11.1

**Comment by VladislavAntonyuk** [comment]
Hmm, it failed on my device. Let me try again

**Comment by darksaviorx** [comment]
Not working here either. Going back to 15.5 for now

**Comment by VladislavAntonyuk** [comment]
> There's no need. V1.1.1 works just fine and you can use macOS up to v11.1

Tried again, and still doesn't work after clean installation

**Comment by julianxhokaxhiu** [comment]
I did try this on VMWare Workstation Player 16 for Windows and worked like a charm at the first try.

Are you trying this on VMWare Workstation Pro?

**Comment by VladislavAntonyuk** [comment]
> I did try this on VMWare Workstation Player 16 for Windows and worked like a charm at the first try.
> 
> Are you trying this on VMWare Workstation Pro?

Yes, I have Workstation Pro

**Comment by mohamedahmedabdelfattah** [comment]
also not working with me after upgrading from workstation pro 15.5.6

**Comment by mohamedahmedabdelfattah** [comment]
> 
> 
> also not working with me after upgrading from workstation pro 15.5.6

so I thought to try again, this time I used process hacker to terminate all the background services and instantly run the unlocker and it worked, MacOS reappeared in the menus and downloading tools again now.

**Comment by julianxhokaxhiu** [comment]
For me the process worked without any kind of issue. Just make sure to run the `Unlocker.exe` after you finish to install VMWare but before you run it at least once. No manual steps required.

Needless to say also, ensure that the process is running as Admin ( and make sure you have the rights to say "Yes" to the UAC prompt ).

**Comment by DX37** [comment]
I'm confirming, that auto-unlocker works, but only if you killed all VMware related services before running Unlocker.exe. Auto-killing doesn't work.

**Comment by VladislavAntonyuk** [comment]
yes, it works. please fix Auto-killing

**Comment by darksaviorx** [comment]
I got it working. The problem was with the enhanced keyboard driver.

**Comment by julianxhokaxhiu** [comment]
> I got it working. The problem was with the enhanced keyboard driver.

This explains why it worked on my end. I didn't install that.

**Comment by utelle** [comment]
After reading the comments to this issue, I had the heart to update my VMware Workstation Pro installation to version 16. 

In principle, `auto-unlocker` worked as expected, but I had to manually kill 2 additional VMware services which locked files that were to be patched:

- VMware VMnet DHCP service (vmnetdhcp.exe)
- VMware NAT Service (vmnat.exe)

It would be nice, if `auto-unlocker` could be adjusted to kill/stop those services automatically, too. Thanks.

**Comment by paolo-projects** [comment]
> After reading the comments to this issue, I had the heart to update my VMware Workstation Pro installation to version 16.
> 
> In principle, `auto-unlocker` worked as expected, but I had to manually kill 2 additional VMware services which locked files that were to be patched:
> 
>     * VMware VMnet DHCP service (vmnetdhcp.exe)
> 
>     * VMware NAT Service (vmnat.exe)
> 
> 
> It would be nice, if `auto-unlocker` could be adjusted to kill/stop those services automatically, too. Thanks.

I will investigate this issue. I have not tried running the unlocker with vmware 16, but if those are the only things to be adjusted it's an easy fix. 

Meanwhile I guess you should run the unlocker right after vmware installation to avoid any service and executable being already running.

**Comment by utelle** [comment]
> I will investigate this issue. I have not tried running the unlocker with vmware 16, but if those are the only things to be adjusted it's an easy fix.

AFAICT the VMware services blocking one of the files unlocker needs to patch are the only issue - at least for me. BTW, I had the same problem when doing VMware minor version upgrading for VMware 15. I'm not sure if that depends on how I installed VMware (actually using defaults) or if there are other reasons.

> Meanwhile I guess you should run the unlocker right after vmware installation to avoid any service and executable being already running.

I upgraded from a previous VMware 15 installation. That is, the VMware services were running when I applied `auto-unlocker --uninstall` - I had to manually stop them so that `auto-unlocker` was able to restore the old files. And after upgrading to VMware 16, the VMware services were immediately restarted by VMware itself, so that I had to stop them again, before performing `auto-unlocker --install`.

Besides that I'm perfectly happy with `auto-unlocker`. My macOSX VM works like a charm.

**Comment by deucalion** [comment]
The patcher works fine for me (once you manually stop all necessary services), there has been only one lingering issue with one of those newly available OS's and their NAT network support (bridged ethernet works fine).

I can't tell why and how, but I suppose I'd have to file an issue for this elsewhere.
(If you know where, please tell me.)

Otherwise: thanks, great patcher, cheers!

**Comment by julianxhokaxhiu** [comment]
> The patcher works fine for me (once you manually stop all necessary services), there has been only one lingering issue with one of those newly available OS's and their NAT network support (bridged ethernet works fine).

If you talk about Linux, this is a known issue since VMWare older releases. The only solution is to use Bridged ethernet.

**Comment by deucalion** [comment]
> > The patcher works fine for me (once you manually stop all necessary services), there has been only one lingering issue with one of those newly available OS's and their NAT network support (bridged ethernet works fine).
> 
> If you talk about Linux, this is a known issue since VMWare older releases. The only solution is to use Bridged ethernet.

No, MacOS (otherwise I wouldn't need the patch 🙈). Thank you tho!
